### ISSUE TITLE
Exception while calling kgtk_quantity_number

### ISSUE THREAD

**Issue by szeke** [usc-isi-i2/kgtk#374]
The following exception occurs while calling `kgtk_quantity_number` in the quantities file from Wikidata. The query succeeds without the `kgtk_quantity_number` so either a bad number of a bug in the function.

The query:
```
!$kypher -i "$QUANTITY" -i "$QUALIFIERS" \
--match 'quantity: (n1)-[l {label: property}]->(n2), qual: (l)-[:P585]->(t)' \
--return 'distinct n1 as node1, property as label, kgtk_quantity_number(n2) as node2, l as id' \
--order-by 'property, node2' \
-o "$TEMP"/quantity.unit.all.tsv
```

The files:
https://drive.google.com/drive/folders/19UbgVbIfe51quxd34NFn3CIcMZgteeEJ?usp=sharing 

The error:
```
[2021-03-19 23:12:00 query]: SQL Translation:
---------------------------------------------
  SELECT DISTINCT graph_17_c1."node1" "_aLias.node1", graph_17_c1."label" "_aLias.label", kgtk_quantity_number(graph_17_c1."node2") "_aLias.node2", graph_7_c2."node1" "_aLias.id"
     FROM graph_17 AS graph_17_c1, graph_7 AS graph_7_c2
     WHERE graph_17_c1."label"=graph_17_c1."label"
     AND graph_7_c2."label"=?
     AND graph_17_c1."id"=graph_7_c2."node1"
     ORDER BY graph_17_c1."label" ASC, "_aLias.node2" ASC
  PARAS: ['P585']
---------------------------------------------
Traceback (most recent call last):
  File "/Users/pedroszekely/opt/anaconda3/envs/kgtk-env/lib/python3.7/site-packages/kgtk-0.5.0-py3.7.egg/kgtk/cli/query.py", line 181, in run
    result = query.execute()
  File "/Users/pedroszekely/opt/anaconda3/envs/kgtk-env/lib/python3.7/site-packages/kgtk-0.5.0-py3.7.egg/kgtk/kypher/query.py", line 730, in execute
    result = self.store.execute(query, params)
  File "/Users/pedroszekely/opt/anaconda3/envs/kgtk-env/lib/python3.7/site-packages/kgtk-0.5.0-py3.7.egg/kgtk/kypher/sqlstore.py", line 235, in execute
    return self.get_conn().execute(*args, **kwargs)
sqlite3.OperationalError: user-defined function raised exception

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/pedroszekely/opt/anaconda3/envs/kgtk-env/lib/python3.7/site-packages/kgtk-0.5.0-py3.7.egg/kgtk/exceptions.py", line 46, in __call__
    return_code = func(*args, **kwargs) or 0
  File "/Users/pedroszekely/opt/anaconda3/envs/kgtk-env/lib/python3.7/site-packages/kgtk-0.5.0-py3.7.egg/kgtk/cli/query.py", line 208, in run
    raise KGTKException(str(e) + '\n')
kgtk.exceptions.KGTKException: user-defined function raised exception

user-defined function raised exception
```

**Comment by chalypso** [comment]
The exception is caused by an integer value overflow from this quantity:

+162000000000000000000000Q122922

which is 77bits long (some amount of Swedish krona) and outside the SQLite 8-byte integer range.  So this actually broke on the SQLite side and not in Python.  I can catch these and map them onto floats for the kgtk_quantity_number accessor, but I have to think about how to handle out-of-range issues for numeric values more generally.

**Comment by CraigMiloRogers** [comment]
As a workaround, converting this integer into a floating point number would probably allow it to be processed in SQLite. I wouldn't want to do this all the time on Wikidata import, because then we couldn't export the Wikidata cleanly (time machine, for example).

**Comment by CraigMiloRogers** [comment]
I could do the conversion in import-wikidata.  It would be nice to have a `kgtk validate` option to detect and/or convert out of range integers, or a `kgtk calc` operation to convert out-of-range integers, but KGTK's data type system doesn't distinguish between integers and other numbers/quantities, which means that my solutions would be more ad-hoc than usual.

**Comment by chalypso** [comment]
I think I fixed it in the spirit of how SQLite handles overflows, will push shortly.  No need to convert on import or validate.

**Comment by chalypso** [comment]
Fixed in latest push to dev (0f91af24..3af7f59b). This now maps large values to floats when possible (e.g., when kgtk_quantity_number' is called), and otherwise clamps values to their range extremes.  This might need further generalization down the road if we want to handle arbitrary precision integers more gracefully or want to actually detect overflows.  I also fixed some other floating point parsing issues, added a test and updated the manual.  Numbers are messy...
